File: src/test/java/com/hileco/cortex/analysis/GraphTest.java
package com.hileco.cortex.analysis;

import com.hileco.cortex.instructions.jumps.JUMP_DESTINATION;
import org.junit.Assert;
import org.junit.Test;

import java.util.List;

public class GraphTest {

    @Test
    public void test() {
        var graph = new Graph(List.of(
                new JUMP_DESTINATION(),
                new JUMP_DESTINATION(),
                new JUMP_DESTINATION()
        ));
        Assert.assertEquals(3, graph.getGraphBlocks().size());
    }
}

File: cortex-analysis/src/test/kotlin/com/hileco/cortex/analysis/GraphTest.kt
package com.hileco.cortex.analysis

import com.hileco.cortex.vm.instructions.jumps.JUMP_DESTINATION
import org.junit.Assert
import org.junit.Test

class GraphTest {
    @Test
    fun test() {
        val graph = Graph(listOf(
                JUMP_DESTINATION(),
                JUMP_DESTINATION(),
                JUMP_DESTINATION()
        ))
        Assert.assertEquals(3, graph.graphBlocks.size)
    }
}